src/operations.jl

```diff
@@ -56,29 +79,83 @@ Base.:(==)(x::ArrayReferenceMeta, y) = false
 
 abstract type AbstractLoopOperation end
 
+"""
+`OperationType` is an `@enum` for classifying supported operations that can appear in
+`@avx` blocks. Type `LoopVectorization.OperationType` to see the different types.
+"""
 @enum OperationType begin
     constant
     memload
     compute
     memstore
     loopvalue
 end
+"An operation setting a variable to a constant value (e.g., `a = 0.0`)" constant
+"An operation setting a variable from a memory location (e.g., `a = A[i,j]`)" memload
+"An operation computing a new value from one or more variables (e.g., `a = b + c`)" compute
+"An operation storing a value to a memory location (e.g., `A[i,j] = a`)" memstore
+"??" loopvalue
 
 # TODO: can some computations be cached in the operations?
 """
+    Operation
+
+A structure to encode a particular action occuring inside an `@avx` block.
+
+# Fields
+
+$(TYPEDFIELDS)
+
+# Example
+
+```jldoctest Operation; filter = r"\\"##.*\\""
+julia> using LoopVectorization
+
+julia> AmulBq = :(for m ∈ 1:M, n ∈ 1:N
+           C[m,n] = zero(eltype(B))
+           for k ∈ 1:K
+               C[m,n] += A[m,k] * B[k,n]
+           end
+       end);
+
+julia> lsAmulB = LoopVectorization.LoopSet(AmulBq);
+
+julia> LoopVectorization.operations(lsAmulB)
+6-element Array{LoopVectorization.Operation,1}:
+ var"##RHS#253" = var"##zero#254"
+ C[m, n] = var"##RHS#253"
+ var"##tempload#255" = A[m, k]
+ var"##tempload#256" = B[k, n]
+ var"##RHS#253" = LoopVectorization.vfmadd_fast(var"##tempload#255", var"##tempload#256", var"##RHS#253")
+ var"##RHS#253" = LoopVectorization.identity(var"##RHS#253")
+```
+Each one of these lines is a pretty-printed `Operation`.
 """
 mutable struct Operation <: AbstractLoopOperation
+    "A unique number assigned to each operation"
     identifier::Int
+    """The name of the variable storing the result of this operation.
+    For `a = val` this would be `:a`. For array assignments `A[i,j] = val` this would be `:A`."""
     variable::Symbol
+    "The size of the result, in bytes. This is not reliable prior to loop reconstruction when types are known."
     elementbytes::Int
+    "The specific operator, e.g., `identity` or `+`"
     instruction::Instruction
+    "The [`OperationType`](@ref) associated with this operation"
     node_type::OperationType
+    "The loop variables this operation depends on"
     dependencies::Vector{Symbol}
+    "??"
     reduced_deps::Vector{Symbol}
```

Comment:
Repeating the `Operation` example:
```julia
julia> using LoopVectorization

julia> AmulBq = :(for m ∈ 1:M, n ∈ 1:N
           C[m,n] = zero(eltype(B))
           for k ∈ 1:K
               C[m,n] += A[m,k] * B[k,n]
           end
       end);

julia> lsAmulB = LoopVectorization.LoopSet(AmulBq);

julia> LoopVectorization.operations(lsAmulB)
6-element Array{LoopVectorization.Operation,1}:
 var"##RHS#265" = var"##zero#266"
 C[m, n] = var"##RHS#265"
 var"##tempload#267" = A[m, k]
 var"##tempload#268" = B[k, n]
 var"##RHS#265" = LoopVectorization.vfmadd_fast(var"##tempload#267", var"##tempload#268", var"##RHS#265")
 var"##RHS#265" = LoopVectorization.identity(var"##RHS#265")

julia> LoopVectorization.loopdependencies.(LoopVectorization.operations(lsAmulB))
6-element Array{Array{Symbol,1},1}:
 [:m, :n]
 [:m, :n]
 [:m, :k]
 [:k, :n]
 [:m, :k, :n]
 [:m, :n]

julia> LoopVectorization.reduceddependencies.(LoopVectorization.operations(lsAmulB))
6-element Array{Array{Symbol,1},1}:
 0-element Array{Symbol,1}
 [:k]
 0-element Array{Symbol,1}
 0-element Array{Symbol,1}
 [:k]
 [:k]
```
It says that operation depends on a reduction over that loop. Here it means that 
`C[m,n] = RHS265` 
and
`var"##RHS#265" = LoopVectorization.identity(var"##RHS#265")`
depend on `m` and `n` directly -- so that they need to be placed inside the `m` and `n` loops -- but that they also depend on a reduction over the `k` loop, and thus have to be placed after the loop.
`var"##RHS#265" = var"##zero#266"`, on the other hand, would be placed before the `k` loop, as it does not depend on it.

`var"##RHS#265" = LoopVectorization.vfmadd_fast(var"##tempload#267", var"##tempload#268", var"##RHS#265")` depends on all three loops, but is a reduction across the `k` loop. I can't immediately recall where or if it matters when `k in loopdependencies(op)`, but it is important for the first two example ops that do not depend directly on `k`.